Azure/sonic-utilities
# Fix Multi-ASIC show specific resursive route by using common parsing function

#### What I did
On Multi-ASIC DUT when the route contains resursive routes in the nexthop information the show ip route handling parsing logic was missing the parsing logic for this case due to it was using its own parsing logic instead of sharing the previous parsing function that already handle the recursive next hop case.  This resulted to hitting an error during show handling similar to following:
```
admin@str2-n3164-acs-1:~$ show ip route -nasic4 -d all 193.11.208.0/25
Routing entry for 193.11.208.0/25
  Known via "bgp", distance 200, metric 0, best
  Last update 13:34:33 ago
Traceback (most recent call last):
  File "/usr/bin/show", line 12, in <module>
    sys.exit(cli())
  File "/usr/lib/python2.7/dist-packages/click/core.py", line 722, in __call__
    return self.main(*args, **kwargs)
  File "/usr/lib/python2.7/dist-packages/click/core.py", line 697, in main
    rv = self.invoke(ctx)
  File "/usr/lib/python2.7/dist-packages/click/core.py", line 1066, in invoke
    return _process_result(sub_ctx.command.invoke(sub_ctx))
  File "/usr/lib/python2.7/dist-packages/click/core.py", line 1066, in invoke
    return _process_result(sub_ctx.command.invoke(sub_ctx))
  File "/usr/lib/python2.7/dist-packages/click/core.py", line 895, in invoke
    return ctx.invoke(self.callback, **ctx.params)
  File "/usr/lib/python2.7/dist-packages/click/core.py", line 535, in invoke
    return callback(*args, **kwargs)
  File "/usr/lib/python2.7/dist-packages/show/main.py", line 1226, in route
    bgp_common.show_routes(args, namespace, display, verbose, "ip")
  File "/usr/lib/python2.7/dist-packages/show/bgp_common.py", line 423, in show_routes
    print_ip_routes(ns_route, filter_by_ip)
  File "/usr/lib/python2.7/dist-packages/show/bgp_common.py", line 159, in print_ip_routes
    str_2_print += ", via {}".format(info[i]['nexthops'][j]['interfaceName'])
KeyError: 'interfaceName'

```

#### How I did it
Changed the code to use the existing parsing function "get_nexthop_info_str()" that already handles recursive nexthop case.
This also improves code reuse.

#### How to verify it
After the fix in place perform similar show command and no longer see the error:

```
admin@str2-n3164-acs-1:~$ show ip route -nasic4 -d all 193.11.208.0/25
Routing entry for 193.11.208.0/25
  Known via "bgp", distance 200, metric 0, best
  Last update 13:34:33 ago
  * 10.0.0.1 (recursive)
  * 10.1.0.1, via PortChannel4004
  * 10.0.0.9 (recursive)
  * 10.1.0.5, via PortChannel4014

 admin@str2-n3164-acs-1:~$
```
Note this change only effects DUT that are Multi-ASIC based. no impact to non-Multi-ASIC platforms.
